### ISSUE TITLE
Actual version of train_dreambooth_lora_sdxl can't train on my photos.

### ISSUE THREAD

**Issue by stilletto** [huggingface/diffusers#6297]
### Describe the bug

I have attempted to train models using several recent versions of the `train_dreambooth_lora_sdxl.py` script found at https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/train_dreambooth_lora_sdxl.py, using my own photos as well as animal photos. Unfortunately, none of the LoRa adaptations reflected what I was trying to teach. For training, I experimented with a range of parameters:

- Learning Rate (LR): from `0.000001568` to `0.0004768`
- Training Steps: from `500` to `6000`

Regardless of the parameters tried, the model failed to learn to replicate even a single image. During retraining attempts, the model either produced incomprehensible noise or simply a black square.

I have tried the code from all commits since December 1st, and none of them resulted in any satisfactory outcomes. However, in the same environment, the `train_text_to_image_lora_sdxl.py` script found at https://github.com/huggingface/diffusers/blob/main/examples/text_to_image/train_text_to_image_lora_sdxl.py produces fairly good results.

### Expected Behavior
The `train_dreambooth_lora_sdxl.py` script should be able to learn from the provided training images and replicate them or produce similar results upon retraining.

### Actual Behavior
- No LoRa adaptations reflect the trained content.
- Models fail to replicate any training images.
- Outputs are either visually corrupted or completely black.

### Reproduction

1. Use photos for training (own photos and animal photos) with the `train_dreambooth_lora_sdxl.py` script.
2. Vary learning rate between `0.000001568` and `0.0004768`.
3. Vary training steps between `500` and `6000`.
4. Observe lack of successful learning or image replication on all commits from December 1st onwards.
example of args I used:
--pretrained_model_name_or_path=/workspace/NightVision_XL   --pretrained_vae_model_name_or_path=/workspace/sdxl-vae-fp16-fix     --resolution=512 --instance_prompt "photo of usrtkuq person" --instance_data_dir=/workspace/image_dataset/    --train_batch_size=1   --gradient_accumulation_steps=1 --gradient_checkpointing   --max_train_steps=5500   --use_8bit_adam   --learning_rate=0.00021 --lr_scheduler="constant" --lr_warmup_steps=0   --mixed_precision="fp16"   --checkpointing_steps=5500   --output_dir="test" --rank 256

### System Info

- `diffusers` version: 0.25.0.dev0
- Platform: Linux-5.15.0-69-generic-x86_64-with-glibc2.31
- Python version: 3.10.11
- PyTorch version (GPU?): 2.0.1 (True)
- Huggingface_hub version: 0.20.1
- Transformers version: 4.34.0
- Accelerate version: 0.25.0
- xFormers version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

**Comment by sayakpaul** [comment]
It could be because of https://github.com/huggingface/diffusers/issues/6087. We have fixed it https://github.com/huggingface/diffusers/pull/6225. Could you recheck with the latest changes and report back?

**Comment by stilletto** [comment]
> It could be because of #6087. We have fixed it #6225. Could you recheck with the latest changes and report back?

Yes, now it can generate my face after 600 steps. Not sure about full functionality yet coz quality not so good. May be for better quality I need to adjust LR and training steps. I run my automated tests with different LR, may be tomorrow get better results.

**Comment by sayakpaul** [comment]
Yeah faces require a bit of experimentation. But that doesn't seem like a bug in the library perse :-)

**Comment by stilletto** [comment]
> Yeah faces require a bit of experimentation. But that doesn't seem like a bug in the library perse :-)

I tried some different steps and learning rate and this script anyway lead to overfit so fast and produce just noise or with lower LR produce images with face almost similar to my but with train_text_to_image_lora_sdxl.py I got better results with less steps. And with SD 1.5 dreambooth script work better too. Not sure where is problem. May the script not work good with 8bitAdam or need loss preservation with better images or some other rank(I tried 4, 8 , 16, 256).

**Comment by sayakpaul** [comment]
Even with that my statement above stands. This definitely seems like a face-specific thing as that script has provided nice results for non-face objects.

**Comment by rvorias** [comment]
I was browsing the issues for things LoRA-quality related; I'm starting to believe something is off since PEFT.

I am keeping everything default except for the dataset (3 photos of a face), and it seems subject finetuning is blursed.

**Comment by sayakpaul** [comment]
> I was browsing the issues for things LoRA-quality related; I'm starting to believe something is off since PEFT.

We did have an issue earlier but it should have been fixed in the latest script. 

If you can help us characterize the issue i.e., point out a version of the script that was working pre-PEFT for you on faces but isn't anymore after PEFT, that would be helpful. Having access to your exact training command and dataset would be helpful, too. Otherwise, it's all a bit vague for us to take any actions here.

**Comment by rvorias** [comment]
I added the lora_alpha params in your fix and it seems to be a ton better o:

**Comment by sayakpaul** [comment]
Is this still a problem?

**Comment by sayakpaul** [comment]
Going to close because of inactivities.
